From d66905d72793e9f010faa8eca8d81555e26848e2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 22 Aug 2026 03:24:26 +0000
Subject: [PATCH 27/36] SimplePartWidget: use QVariant::value<QModelIndex>()
 instead of toModelIndex()

QVariant::toModelIndex() is a Qt 5.5+ convenience method (added for
QML). QModelIndex has always been a registered metatype, so
.value<QModelIndex>() is the portable equivalent, available
identically on Qt4 and every Qt5 version. Only instance in the tree.
---
 src/Gui/SimplePartWidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Gui/SimplePartWidget.cpp b/src/Gui/SimplePartWidget.cpp
index e9233be3..b5e017e5 100644
--- a/src/Gui/SimplePartWidget.cpp
+++ b/src/Gui/SimplePartWidget.cpp
@@ -236,7 +236,7 @@ void SimplePartWidget::slotDownloadPart()
 
 void SimplePartWidget::slotDownloadMessage()
 {
-    QModelIndex index = m_partIndex.data(Imap::Mailbox::RolePartMessageIndex).toModelIndex();
+    QModelIndex index = m_partIndex.data(Imap::Mailbox::RolePartMessageIndex).value<QModelIndex>();
     Imap::Network::FileDownloadManager *manager = new Imap::Network::FileDownloadManager(this, m_netAccessManager, index);
     connect(manager, SIGNAL(fileNameRequested(QString*)), this, SLOT(slotFileNameRequested(QString*)));
     connect(manager, SIGNAL(transferError(QString)), m_messageView, SIGNAL(transferError(QString)));
-- 
2.43.0

